### DISCUSSION TITLE
Building Docker image for AWS Lambda

### DISCUSSION THREAD

**Discussion by bryanjswift** [moonrepo/moon#565] [Q&A] [answered]
One of the artifacts which gets built in my project is a Docker image to be executed with AWS Lambda. When I tried porting this build process to use moon I encountered these errors

```
#23 0.707 ./node_modules/.bin/moon: /lib64/libc.so.6: version `GLIBC_2.29' not found (required by ./node_modules/.bin/moon)
#23 0.707 ./node_modules/.bin/moon: /lib64/libc.so.6: version `GLIBC_2.28' not found (required by ./node_modules/.bin/moon)
#23 0.707 ./node_modules/.bin/moon: /lib64/libc.so.6: version `GLIBC_2.27' not found (required by ./node_modules/.bin/moon)
```

The Lambda base images for node appear to have GLIBC_2.26 installed. I'm sure there are many ways to work around this, one I'm considering is attempting to build `moon` inside the builder image so it links against the available library version.

My question here is if it's possible or reasonable to build releases against this older glibc version?

**Comment by milesj** [comment, marked as answer]
We have some info on this problem here: https://moonrepo.dev/docs/faq#how-to-resolve-the-version-glibc_xxx-not-found-error

Unfortunately the glibc we compile with is based on the GitHub actions runners, which we don't have much control over. We haven't had the time or resources to look into alternative solutions yet.

**Reply by bryanjswift** [reply]
Copy, no problem. I suspect there are other ways for me to resolve the issue for my team’s use case. I’ll definitely take a look at the documentation you pointed me at as well

**Reply by bryanjswift** [reply]
I ended up using a different amazon Linux based build image and, at least in my case, that is working fine.
